packages/manager/apps/web/client/app/hosting/cdn/shared/hosting-cdn-shared-settings.routing.js

```diff
@@ -0,0 +1,66 @@
+import includes from 'lodash/includes';
+
+import template from './leave-confirm-settings/leave/hosting-shared-leave-settings.html';
+import controller from './leave-confirm-settings/leave/hosting-shared-leave-settings.controller';
+import CdnSharedSettingsController from './hosting-cdn-shared-settings.controller';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('app.hosting.cdn.shared', {
+    url: '/shared/settings/:domainName',
+    params: {
+      domainName: null,
+      domain: null,
+    },
+    component: 'hostingCdnSharedSettings',
+    resolve: {
+      serviceName: /* @ngInject */ ($transition$) =>
+        $transition$.params().productId,
+
+      goBack: /* @ngInject */ ($state) => () => $state.go('^'),
+    },
+
+    onExit: ($transition$) => {
```

Comment:
```suggestion
    onExit: (transition) => {
```

Reply:
i'am not sure we can use transition instead of $transition$.
According the docs it use $transition$.

Reply:
I confirm, i tested and i got issue by using `transition` instead of `$transition$`.

Reply:
> According the docs it use $transition$.

According to the documentation, `onExit` should be a `TransitionStateHookFn` (https://ui-router.github.io/ng1/docs/latest/interfaces/state.statedeclaration.html#onexit).
And `TransitionStateHookFn`'s signature is `(transition: Transition, state: StateDeclaration)` (https://ui-router.github.io/ng1/docs/latest/interfaces/transition.transitionstatehookfn.html).

So it should works with `onExit: (transition, state)=> {}`.

But, I agree, it doesn't work.

Why ? because we are using `babel-plugin-angularjs-annotate` (based on `ng-annotate`) and `ng-annotate` annotates ui-router `$stateProvider` and `$urlRouteProvider` (including state declaration and hooks, cf https://github.com/schmod/babel-plugin-angularjs-annotate/blob/master/IMPLICIT.md#declaration-forms), so it tries to inject `transition` but don't find it.

In a previous version of UiRouter `onExit` wait a function with injections, but it's no more the case, and I think `ng-annotate` is not up to date with `@uirouter/angularjs@1.x.x`

Reply:
thanks for info.